modules/system/tests/classes/CombineAssetsTest.php

```diff
@@ -104,4 +104,28 @@ public function testResetCache()
         $combiner = CombineAssets::instance();
         $this->assertNull($combiner->resetCache());
     }
+
+    public function testAliasesResolving()
+    {
+        $combiner = CombineAssets::instance();
+
+        $combiner->registerAlias('css1', '~/modules/system/tests/fixtures/themes/test/assets/css/style1.css');
+        $combiner->registerAlias('css2', '~/modules/system/tests/fixtures/themes/test/assets/css/style2.css');
+        $combiner->registerAlias('js1', '~/modules/system/tests/fixtures/themes/test/assets/js/script1.js');
+        $combiner->registerAlias('js2', '~/modules/system/tests/fixtures/themes/test/assets/js/script2.js');
+
+        $url = $combiner->combine(
+            ['@js1', '@js2'],
+            base_path() . '/modules/system/tests/fixtures/themes/test'
+        );
+        $this->assertNotNull($url);
+        $this->assertRegExp('/\w+[-]\d+/i', $url); // Must contain hash-number
+
+        $url = $combiner->combine(
+            ['@css1', '@css2'],
+            base_path() . '/modules/system/tests/fixtures/themes/test'
+        );
+        $this->assertNotNull($url);
+        $this->assertRegExp('/\w+[-]\d+/i', $url); // Must contain hash-number
+    }
```

Comment:
What happens if both CSS & JS assets are present? What about if an alias that is registered for both is used? Can you add the following test cases:

- 0 aliases
- 1 css
- 1 js
- 2 css 1 js
- 1 css 2 js
- 2 css 2 js
- 1 combined
- 1 combined 1 css
- 1 combined 1 js
- 1 combined 1 js 1 css

Reply:
> What happens if both CSS & JS assets are present?

Both will be combined, into a file with the extension of the dominant stack count (if 2 js and 1 css, it will return a `.js` file, not sure for 1 vs 1), there is actually absolutely no filter in the combiner to exclude programmer's errors.

> What about if an alias that is registered for both is used?

With the actual logic, the alias which is registered after is dominant.  
As of today, that's what happens `framework.extras` js alias which overrides the css alias. If you explicitly ask the combiner for a css file (passing a css not-aliased file into the combiner), the css alias will be correctly used.  

But in all of the other cases (equal css/js stacks counts and js dominant), the js alias will be used.

I will add these tests later in the day. Just to be sure what do you call "1 combined"?

Reply:
1 combined means an alias that refers to both JS & CSS, i.e. `framework.extras`

Reply:
I started writing the tests but I'm not sure that we are on the same line on what some tests should return:
- mixed assets type (ie 1 or more css + 1 or more js). As of today it combines "properly" and returns a mixed css/js file, should we keep this behavior without any warning?
- 1 combined and 1 combined 1 js 1 css. As of today the combined asset will be the js one due to js aliases dominance (detailed here: https://github.com/wintercms/winter/pull/634#discussion_r942267173), and I have no way into the combine method to select the css if I would like to.

I still will finish writing tests as draft so we can easily modify them together before going further in refactorization.

Reply:
I updated my comment with an initial thought of what should happen for each but I'm quite tired right now and can't really think about it too hard, so perhaps @bennothommo can have a read of this PR and add his thoughts.

Reply:
To me:
2 css 1 js - css, js not added > agreed, with a warning to notice the dev why the js is not added
1 css 2 js - js, css not added > agreed, with a warning to notice the dev why the css is not added
2 css 2 js - ? > error: can't guess the wanted asset's type
1 combined 1 css - css > ok
1 combined 1 js - js > ok
1 combined 1 js 1 css - js > why js? to me it's the same as 2 css 2 js: error, we can't guess the wanted asset's type